pkg/controller/podautoscaler/metrics/rest_metrics_client.go

```diff
@@ -79,13 +79,15 @@ func (c *resourceMetricsClient) GetResourceMetric(resource v1.ResourceName, name
 		podSum := int64(0)
 		missing := len(m.Containers) == 0
 		for _, c := range m.Containers {
-			resValue, found := c.Usage[v1.ResourceName(resource)]
-			if !found {
-				missing = true
-				klog.V(2).Infof("missing resource metric %v for container %s in pod %s/%s", resource, c.Name, namespace, m.Name)
-				break // containers loop
+			if container == "" || container == c.Name {
+				resValue, found := c.Usage[resource]
+				if !found {
+					missing = true
+					klog.V(2).Infof("missing resource metric %v for container %s in pod %s/%s", resource, c.Name, namespace, m.Name)
```

Comment:
Fix the log line for when both cases: pod and container metrics are requested. Currently only handles container case.